smartdispatch/tests/test_smartdispatch_script.py

```
@@ -0,0 +1,49 @@
+import unittest
+from smartdispatch import smartdispatch_script
+import subprocess
+from mock import patch
+import tempfile as tmp
+import shutil
+
+
+class TestSmartScript(unittest.TestCase):
+
+    def setUp(self):
+        self._base_dir = tmp.mkdtemp()
+        smartdispatch_script.LOGS_FOLDERNAME = self._base_dir
+
+    def tearDown(self):
+        shutil.rmtree(self._base_dir)
+
+    def test_gpu_check(self):
+
+        argv = ['-x', '-g', '2', '-G', '1', '-C', '1', '-q', 'random', '-t', '10' ,'launch', 'echo', 'testing123']
+
+        with self.assertRaises(SystemExit) as context:
+            smartdispatch_script.main(argv=argv)
+
+        self.assertTrue(context.exception.code, 2)
+
+    def test_cpu_check(self):
+
+        argv = ['-x', '-c', '2', '-C', '1', '-G', '1', '-t', '10', '-q', 'random', 'launch', 'echo', 'testing123']
+
+        with self.assertRaises(SystemExit) as context:
+            smartdispatch_script.main(argv=argv)
+
+        self.assertTrue(context.exception.code, 2)
+
+    @patch('subprocess.check_output')
+    def test_launch_job_check(self, mock_check_output):
+
+        mock_check_output.side_effect = subprocess.CalledProcessError(1, 1, "A wild error appeared!")
+        argv = ['-t', '0:0:1', '-G', '1', '-C', '1', '-q', 'random', 'launch', 'echo', 'testing123']
+
+        try:
+            with self.assertRaises(SystemExit) as context:
+                smartdispatch_script.main(argv=argv)
+
+                self.assertTrue(context.exception.code, 2)
+
+        except subprocess.CalledProcessError:
+            self.fail("smartdispatch_script.main() raised CalledProcessError unexpectedly!")
```

Do you know if self.fail will keep the stack trace from CalledProcessError? I know it is mocked so there won't be interesting information in the message itself, but the stacktrace would point to the unexpected call to check_output.

---

No it doesn't. Gonna add the stack trace to the assert message.

---

I think it would be better to reraise the error with the message changed than adding the stacktrace as a string to the message (if I understand correctly what you intend).

I would use [six.reraise](https://pythonhosted.org/six/#six.reraise). You can look [here](https://github.com/mila-udem/blocks/blob/master/blocks/utils/utils.py#L96) for a usage example.